pkg/controller/account/account_controller.go

```diff
@@ -849,18 +850,24 @@ func CreateEC2Instance(client awsclient.Client) (string, error) {
 			MinCount:     aws.Int64(1),
 			MaxCount:     aws.Int64(1),
 		})
-		if runErr == nil {
-			instanceID = *runResult.Instances[0].InstanceId
-			break
+
+		// Return on unexpected errors
+		if runErr != nil {
+			aerr, ok := runErr.(awserr.Error)
```

Comment:
I'm reading through some other parts of the operator and its getting terse to read. Could you add comments into your PR, especially in this block?